plugins/pulp_python/plugins/distributors/steps.py

```diff
@@ -60,67 +59,125 @@ def process_main(self):
+            for project_name, packages in projects.items():
+                element = ElementTree.SubElement(body, 'a', {'href': project_name})
+                element.text = project_name
                 ElementTree.SubElement(body, 'br')
-                PublishMetadataStep._create_package_index(name, simple_path, packages)
+                PublishMetadataStep._create_project_index(project_name, simple_path, packages)
 
             index.write(ElementTree.tostring(html, 'utf8'))
 
     @staticmethod
-    def _create_package_index(name, simple_path, packages):
+    def _create_project_index(project_name, simple_path, packages):
         """
-        Create a folder in simple_path named after the package, and then create a package index
-        file in that path.
+        Create a project index.html in a project subdirectory that contains references to each
+        package of the project.
 
-        :param name:        The name of the package
-        :type  name:        basestring
-        :param simple_path: The path to the simple/ publish folder
+        :param project_name: The name of the project
+        :type  project_name: basestring
+        :param simple_path: The path to the simple publish directory
         :type  simple_path: basestring
-        :param packages:    A list of dictionaries of the form
-                            {'version': VERSION, 'filename': FILENAME, 'checksum': MD5SUM,
-                             'checksum_type': TYPE}.
-        :type  packages:    list
-        """
-        # Now we need a subfolder for this package with its own index
-        package_path = os.path.join(simple_path, name)
-        os.makedirs(package_path)
-        package_index_path = os.path.join(package_path, 'index.html')
+        :param packages: A list of packages
+        :type  packages: list of pulp_python.plugins.models.Package
+        """
+        project_path = os.path.join(simple_path, project_name)
+        os.makedirs(project_path)
+        package_index_path = os.path.join(project_path, 'index.html')
         with open(package_index_path, 'w') as package_index:
             html = ElementTree.Element('html')
             head = ElementTree.SubElement(html, 'head')
             title = ElementTree.SubElement(head, 'title')
-            title.text = 'Links for %s' % name
+            title.text = 'Links for %s' % project_name
             ElementTree.SubElement(head, 'meta', {'name': 'api-version', 'value': '2'})
             body = ElementTree.SubElement(html, 'body')
             heading = ElementTree.SubElement(body, 'h1')
             heading.text = title.text
             for package in packages:
-                href = '../../%s#%s=%s' % (_get_package_path(name, package['filename']),
-                                           package['checksum_type'], package['checksum'])
+                href = '../../packages/%s' % package.checksum_url
```

Comment:
If you opt to add a method to the model to build this url, you can replace it here I think.